src/win/basename_patch.h

```diff
@@ -0,0 +1,22 @@
+/* Authors:
+ * 2021 Tamás Hakkel <hakkelt@gmail.com>
+ */
+
+#ifndef BASENAME_PATCH_H
+#define BASENAME_PATCH_H
+
+#include <stdlib.h>
+#include <string.h>
+
+char* win_basename(const char *path);
+
+char* win_basename(const char *path) {
+    char* substr = strrchr(path, '\\');
+    if (NULL == substr)
+        substr = strrchr(path, '/');
+    return NULL == substr ? path : substr + 1;
+}
+
+#define basename(path) win_basename(path)
```

Comment:
Although `basename` is part of MinGW and it is mapped to some Windows API calls, calls to this function threw segmentations faults. Therefore, I replaced it with a simplified implementation.